openedx/core/djangoapps/config_model_utils/tests/test_appconfig.py

```diff
@@ -0,0 +1,176 @@
+        self.make_config_object(org_course=self.org1_course, slug="config-5")
+        self.make_config_object(context_key=self.org1_course_run, slug="config-6")
+
+    def test_disallow_multiple_levels(self):
+        with self.assertRaises(ValidationError) as context:
+            self.model.objects.create(
+                org="Test",
+                org_course="Test/Course",
+            ).clean()
+
+        self.assertEqual(
+            context.exception.message,
+            "Configuration may not be specified at more than one level at once."
+        )
+
+    def test_disallow_slug_reuse_across_levels(self):
+        self.model.objects.create(org="Test", slug="test-slug").clean()
+
+        with self.assertRaises(ValidationError) as context:
+            self.model.objects.create(site=SiteFactory.create(), slug="test-slug").clean()
+
+        self.assertEqual(
+            context.exception.message,
+            "This configuration slug is already used by another configuration."
+        )
+
+        with self.assertRaises(ValidationError) as context:
+            self.model.objects.create(org="Test2", slug="test-slug").clean()
+
+        self.assertEqual(
+            context.exception.message,
+            "This configuration slug is already used by another configuration."
+        )
+
+        with self.assertRaises(ValidationError) as context:
+            self.model.objects.create(org_course="Test+Course", slug="test-slug").clean()
+
+        self.assertEqual(
+            context.exception.message,
+            "This configuration slug is already used by another configuration."
+        )
+
+    def test_allow_same_slug_at_same_level(self):
+        self.make_config_object(org=self.org1, slug="test-slug").clean()
+        self.make_config_object(org=self.org1, slug="test-slug").clean()
+
+        self.make_config_object(org_course=self.org1_course, slug="test-course-slug").clean()
+        self.make_config_object(org_course=self.org1_course, slug="test-course-slug").clean()
+
+    def test_get_current(self):
+        config1 = self.make_config_object(org=self.org1, slug="test-slug")
+        config2 = self.make_config_object(org=self.org1, slug="test-slug")
+        # config2 should override config1.
+        self.assertNotEqual(self.model.current(context_key=self.org1_course_run, slug="test-slug"), config1)
+        self.assertEqual(self.model.current(context_key=self.org1_course_run, slug="test-slug"), config2)
+        config2_id = config2.id
+        config2.save()
+        # The current model should not have the same id as the old config2 since saving generates a new id
+        # and doesn't modify the existing object
+        self.assertNotEqual(self.model.current(context_key=self.org1_course_run, slug="test-slug").id, config2_id)
+
+    @ddt.data(
+        # site, org, org_course, context_key, slug_keys
+        (1, None, None, None, [1]),
+        (2, None, None, None, [2]),
```

Comment:
Should we use `self.site1.id` and `self.site2.id`?

Reply:
This is a decorator so exists before self is available. I will look into this. This test seemed to pass locally, not sure what the issue is here.